File: Elearn-DataTier/src/main/java/via/dk/elearn/models/LectureVote.java
package via.dk.elearn.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@IdClass(LectureVoteId.class)
public class LectureVote {

    @Id
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private User user;

    @Id
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "lecture_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Lecture lecture;

    @Column(nullable = false)
    private boolean upvote;//true = upvote, false = downvote
}

File: Elearn-ApplicationTier/Elearn.Shared/Models/LectureVote.cs
namespace Elearn.Shared.Models;

public class LectureVote
{
    public User User { get; set; }
    public Lecture Lecture { get; set; }
    public bool Upvote { get; set; }

    public LectureVote(User user, Lecture lecture, bool upvote)
    {
        User = user;
        Lecture = lecture;
        Upvote = upvote;
    }
}